src/components/Products.js

```diff
@@ -27,6 +26,39 @@ function Proudects() {
       .catch((err) => setProducts([]));
   }, [setProducts]);
 
+  if (products) {
+    for (let i = 0; i <= 19; i++) {
+      products[i]["quantity"] = 1;
+    }
+    for (let i = 0; i <= 19; i++) {
+      products[i]["color"] = "black";
+    }
+  }
+
+  function addProducTtoCart() {
+    if (cartProduct) {
+      setCartProduct((product) =>
+        product?.map((item) =>
+          products[url ? url - 1 : 1].id === item.id
+            ? { ...item, quantity: item.quantity + 1 }
+            : item
+        )
+      );
+    }
+    setCartProduct((prev) => [
+      ...prev,
+      {
+        id: products[url ? url - 1 : 1].id,
+        image: products[url ? url - 1 : 1].image,
```

Comment:
This function is repeated, add it to utils files and pass the state and setState